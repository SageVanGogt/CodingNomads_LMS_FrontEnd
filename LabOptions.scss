.LabOptions_select {
  width: 240px;
  height: 30px;
  margin-top: 10px;
  font-size: 12px;
}

.LabOptions_delete {
  height: 25px;
}
