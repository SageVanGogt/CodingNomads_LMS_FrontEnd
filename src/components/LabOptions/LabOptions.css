.LabOptions_select {
  width: 240px;
  height: 20px;
  font-size: 12px; }

.LabOptions_delete {
  height: 15px;
  border: none;
  margin-left: 10px;
  cursor: pointer; }
