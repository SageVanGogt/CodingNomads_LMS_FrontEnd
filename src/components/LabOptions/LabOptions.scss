.LabOptions_select {
  width: 240px;
  height: 20px;
  margin-top: 5px;
  font-size: .9rem;
}

.LabOptions_delete {
  background-color: transparent;
  height: 15px;
}
