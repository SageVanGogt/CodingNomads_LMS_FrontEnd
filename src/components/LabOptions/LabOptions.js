import React from 'react';
import './LabOptions';
import PropTypes from 'prop-types';

export const LabOptions = ({ labs, handleSelectLab, deleteLab, id, labSelected }) => {
  const labOptions = labs.map((lab, index) => {
    return (
      <option 
        key={`lab-${index}`} 
        name="lab"
        title={lab.description}
        value={lab.id}
      >
        {lab.name}
      </option>
    );
  });

  return (
    <div>
      <select 
        name="" 
        value={labSelected && labSelected.id} 
        onChange={(event) => handleSelectLab(event)}
      >
        <option value='' disabled selected></option>
        { labOptions }
      </select>
      <button 
        onClick={(event) => deleteLab(event, id)}
      >
        del
      </button>
    </div>
  );
};

LabOptions.propTypes = {
  handleSelectLab: PropTypes.func,
  labs: PropTypes.array
};