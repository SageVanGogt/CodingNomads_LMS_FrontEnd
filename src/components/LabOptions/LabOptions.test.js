import React from 'react';
import { shallow } from 'enzyme';
import { LabOptions } from './LabOptions';

describe('LabOptions', () => {
  it('matches the snapshot if there is a selectedLab', () => {
    const mockProps = {
      labs: [],
      handleSelectLab: () => {},
      deleteLab: () => {},
      id: 1,
      labSelected: { id: 1 }
    }
    const wrapper = shallow(<LabOptions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot if there is not a selectedLab', () => {
    const mockProps = {
      labs: [],
      handleSelectLab: () => {},
      deleteLab: () => {},
      id: 1,
    }
    const wrapper = shallow(<LabOptions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();

  })

})
