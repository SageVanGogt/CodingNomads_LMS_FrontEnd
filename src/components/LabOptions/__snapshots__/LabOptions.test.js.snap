// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabOptions matches the snapshot if a array of labs is passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteLab={[Function]}
    handleSelectLab={[Function]}
    id={1}
    labSelected={
        Object {
            "id": 1,
          }
    }
    labs={
        Array [
            Object {
              "id": 1,
              "topic": "lab topic",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="LabOptions_select"
          name=""
          onChange={[Function]}
          value={1}
>
          <option
                    disabled={true}
                    selected={true}
                    value=""
          />
          <option
                    name="lab"
                    title={undefined}
                    value={1}
          />
</select>,
        <img
          className="LabOptions_delete"
          onClick={[Function]}
          src="/button.png"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              selected={true}
              value=""
/>,
            Array [
              <option
                name="lab"
                title={undefined}
                value={1}
/>,
            ],
          ],
          "className": "LabOptions_select",
          "name": "",
          "onChange": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "lab-0",
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "name": "lab",
              "title": undefined,
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "LabOptions_delete",
          "onClick": [Function],
          "src": "/button.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="LabOptions_select"
            name=""
            onChange={[Function]}
            value={1}
>
            <option
                        disabled={true}
                        selected={true}
                        value=""
            />
            <option
                        name="lab"
                        title={undefined}
                        value={1}
            />
</select>,
          <img
            className="LabOptions_delete"
            onClick={[Function]}
            src="/button.png"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                disabled={true}
                selected={true}
                value=""
/>,
              Array [
                <option
                  name="lab"
                  title={undefined}
                  value={1}
/>,
              ],
            ],
            "className": "LabOptions_select",
            "name": "",
            "onChange": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "lab-0",
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "name": "lab",
                "title": undefined,
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "LabOptions_delete",
            "onClick": [Function],
            "src": "/button.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabOptions matches the snapshot if there is a selectedLab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteLab={[Function]}
    handleSelectLab={[Function]}
    id={1}
    labSelected={
        Object {
            "id": 1,
          }
    }
    labs={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="LabOptions_select"
          name=""
          onChange={[Function]}
          value={1}
>
          <option
                    disabled={true}
                    selected={true}
                    value=""
          />
</select>,
        <img
          className="LabOptions_delete"
          onClick={[Function]}
          src="/button.png"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              selected={true}
              value=""
/>,
            Array [],
          ],
          "className": "LabOptions_select",
          "name": "",
          "onChange": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "LabOptions_delete",
          "onClick": [Function],
          "src": "/button.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="LabOptions_select"
            name=""
            onChange={[Function]}
            value={1}
>
            <option
                        disabled={true}
                        selected={true}
                        value=""
            />
</select>,
          <img
            className="LabOptions_delete"
            onClick={[Function]}
            src="/button.png"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                disabled={true}
                selected={true}
                value=""
/>,
              Array [],
            ],
            "className": "LabOptions_select",
            "name": "",
            "onChange": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "LabOptions_delete",
            "onClick": [Function],
            "src": "/button.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabOptions matches the snapshot if there is not a selectedLab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteLab={[Function]}
    handleSelectLab={[Function]}
    id={1}
    labs={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="LabOptions_select"
          name=""
          onChange={[Function]}
          value={undefined}
>
          <option
                    disabled={true}
                    selected={true}
                    value=""
          />
</select>,
        <img
          className="LabOptions_delete"
          onClick={[Function]}
          src="/button.png"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              selected={true}
              value=""
/>,
            Array [],
          ],
          "className": "LabOptions_select",
          "name": "",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "LabOptions_delete",
          "onClick": [Function],
          "src": "/button.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="LabOptions_select"
            name=""
            onChange={[Function]}
            value={undefined}
>
            <option
                        disabled={true}
                        selected={true}
                        value=""
            />
</select>,
          <img
            className="LabOptions_delete"
            onClick={[Function]}
            src="/button.png"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                disabled={true}
                selected={true}
                value=""
/>,
              Array [],
            ],
            "className": "LabOptions_select",
            "name": "",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "LabOptions_delete",
            "onClick": [Function],
            "src": "/button.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
